src/library_pipefs.js

```diff
@@ -0,0 +1,189 @@
+mergeInto(LibraryManager.library, {
+  $PIPEFS__postset: '__ATINIT__.push(function() { PIPEFS.root = FS.mount(PIPEFS, {}, null); });',
+  $PIPEFS__deps: ['$FS'],
+  $PIPEFS: {
+    BUCKET_BUFFER_SIZE: 1024 * 8, // 8KiB Buffer
+    mount: function(mount) {
+      return FS.createNode(null, '/', {{{ cDefine('S_IFDIR') }}} | 511 /* 0777 */, 0);
+    },
+    createPipe: function () {
+      var pipe = {
+        buckets: []
+      };
+
+      pipe.buckets.push({
+        buffer: new Uint8Array(PIPEFS.BUCKET_BUFFER_SIZE),
+        offset: 0,
+        roffset: 0
+      });
+
+      var rName = PIPEFS.nextname();
+      var wName = PIPEFS.nextname();
+      var rNode = FS.createNode(PIPEFS.root, rName, {{{ cDefine('S_IFIFO') }}}, 0);
+      var wNode = FS.createNode(PIPEFS.root, wName, {{{ cDefine('S_IFIFO') }}}, 0);
+
+      rNode.pipe = pipe;
+      wNode.pipe = pipe;
+
+      var readableStream = FS.createStream({
+        path: rName,
+        node: rNode,
+        flags: FS.modeStringToFlags('r'),
+        seekable: false,
+        stream_ops: PIPEFS.stream_ops
+      });
+      rNode.stream = readableStream;
+
+      var writableStream = FS.createStream({
+        path: wName,
+        node: wNode,
+        flags: FS.modeStringToFlags('w'),
+        seekable: false,
+        stream_ops: PIPEFS.stream_ops
+      });
+      wNode.stream = writableStream;
+
+      return [ readableStream.fd, writableStream.fd ];
```

Comment:
may as well return an object here, so you don't need to use 0 and 1 to index, but can use proper names.